service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/twin/Twin.java

```diff
@@ -109,29 +109,35 @@ public static Twin fromJson(String json)
         }
 
         // Desired properties
-        twin.getDesiredProperties().setVersion(twinState.getDesiredProperties().getVersion());
```

Comment:
We were dereferencing twinState.getDesiredProperties() without null checking it first. Now we will null check first.